### ISSUE TITLE
RightCurly doesn't take comments into consideration

### ISSUE THREAD

**Issue by xenoterracide** [checkstyle/checkstyle#14278]
**How it works Now:**

```
[ERROR] /Users/ccushing/IdeaProjects/cg-copperhead/cg-copperhead-customer/src/main/java/com/kroger/cg/customer/Scratch.java:8:9: '}' at column 9 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurly]
```
```
package com.kroger.cg.customer;

class Scratch {

    public static void main(String[] args) {
        if (args == null) {
            System.out.println("null");
        }
        // my comment
        else if (args.length == 1) {
            System.out.println("one");
        } else if (args.length > 1) {
            System.out.println("greater than one");
        }
    }
}
```

I think this is the correct minimal subset of our config
```xml
<?xml version="1.0" ?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="RightCurly" />
    </module>
</module>
```

versions 10.1

**Is your feature request related to a problem? Please describe.**

We have comments on some of our conditionals, after reading the documentation for `RightCurly` I don't see any way of configuring its correct behavior to accomodate those comments. Longterm answer is to move them to after the brace.

**Describe the solution you'd like**

Some way of providing a configuration for how a comment in the middle behaves differently. I want some way to make this class pass without removing the configuration completely or adding line item ignores.

```bash
expected/suggested cli output
```

**Additional context**

This has even more code, but it's less "clean"

https://github.com/jhipster/prettier-java/issues/637

**Comment by nrmancuso** [comment]
Link to check documentation: https://checkstyle.sourceforge.io/checks/blocks/rightcurly.html#RightCurly

I am not clear on exactly what the request is here, are you looking to introduce a new property to allow comments? Or should the check handle this by default?

>Longterm answer is to move them to after the brace.

Yes, agreed here. I typically request the same when this comes up in review.

I would recommend to suppress such violations via XPath using one of our filters at https://checkstyle.sourceforge.io/filters/index.html. The expression to match this should look something like `//RCURLY[./../../SINGLE_LINE_COMMENT]`.

**Comment by xenoterracide** [comment]
> I am not clear on exactly what the request is here, are you looking to introduce a new property to allow comments? Or should the check handle this by default?

I would like a new property to allow this, it surprises me that it can't be done now. I mean I wouldn't have chosen to write in the style but it doesn't feel like it bad code... and not having a property to allow it means it's harder to enforce the good code.

**Comment by romani** [comment]
Suppression filter is always way to avoid  violation on code that is unusual.
Creation of new property is heavy process of designing and considering numerous edge cases, someone needs to do this. 

We can not put approval on this issue for now, but if someone do a work and show that new property is good we will merge a fix. We put approval labels only if sure is fix.
